src/services/offer.js

```diff
@@ -262,6 +309,30 @@ class OfferService {
         return constraints.length ? { "$and": constraints } : {};
     }
 
+    /**
+     * Encodes a query token, by taking the offer's id and FTS score (if present) and decoding in safe url base64
+     * @param {*} offer
+     */
+    encodeQueryToken(offer) {
+        return base64url.encode(JSON.stringify({
+            id: offer._id,
+            score: offer.score || offer._doc?.score
```

Comment:
Why the need for this OR? You shouldn't be accessing `_doc`

Reply:
If I remember correctly, the query returned by `find()` includes the score attribute. However, `aggregate()`returns an instance of the `Aggregate` class, in which I could only access the score by checking `_doc`

Reply:
And I couldn't filter by textScore in a normal `find()`

Reply:
the encode function should not depend on internal implementations, it should receive the `id` and `score` directly, and build the token